### ISSUE TITLE
长短不同的句子，batch合成的语音长度是一样的

### ISSUE THREAD

**Issue by lanyuer** [2noise/ChatTTS#637]
第一个句子10个字，第二个100字的话，合成出来两句话都是十几秒，第一句后面都是静音

可能是什么地方有问题呢？

**Comment by fumiama** [comment]
这是批量解码导致的，解决方法是尽量使每句话长度相同，最后手动删去末尾的静音（全0数组）。

**Comment by ai408** [comment]
@fumiama 请问下有什么好的办法删除每句话末尾的静音吗？

**Comment by ai408** [comment]
@fumiama 谢谢，很好的思路。不过实际我发现尾部并不是0，而是很小的数字，比如1e-8等。

**Comment by overman1** [comment]
> @fumiama 谢谢，很好的思路。不过实际我发现尾部并不是0，而是很小的数字，比如1e-8等。
如下是我的方式，初步验证是OK的，欢迎大家验证：
wavs = chat.infer(inputs_text )
# 设定阈值
threshold = 1e-7  

processed_wavs = []
for wav in wavs:
    i = len(wav) - 1
    while i >= 0 and abs(wav[i]) < threshold:
        i -= 1
    processed_wavs.append(wav[:i+1])  # 保留直到发现大于阈值的元素为止的所有元素

wavs_2d = [torch.tensor(wav[None, :]) for wav in processed_wavs]  
finally_wavs = torch.cat(wavs_2d, dim=1)  # finally_wavs现在是一个形状为(1, total_samples)的张量，适合单声道音频
torchaudio.save("output.wav",finally_wavs, 24000)
